x-pack/plugins/monitoring/server/lib/elasticsearch/nodes/get_nodes/get_paginated_nodes.ts

```diff
@@ -61,9 +63,10 @@ export async function getPaginatedNodes(
   const nodes: Node[] = await getNodeIds(req, { clusterUuid }, size);
 
   // Add `isOnline` and shards from the cluster state and shard stats
-  const clusterState = clusterStats?.cluster_state ?? { nodes: {} };
+  const clusterState = clusterStats?.cluster_state ??
+    clusterStats?.elasticsearch?.cluster?.stats?.state ?? { nodes: {} };
```

Comment:
Stack Monitoring has a lot of code like this and it makes me uncomfortable. 
I feel like it doesn't explain on it's own why the code path is like this. 

Looking at your comment in your issue, I now understand that it has something to do with not grabbing the data from the ECS path but the code makes no such mention. It just says "if it's undefined here, check here, else empty". 

I wonder if we can extract this to a function instead that explains why it grabs the data from where and in which case it follows what route?

Reply:
Unfortunately this is all over the code base due to the ECS path changes and it can be expected in almost every ES query response, so I'd prefer to keep it the same in this PR.  But perhaps we can open another PR that creates a mapping of ECS to legacy paths all in one place (perhaps by product mapping) and a helper function and update all the queries to use that.

Reply:
That would be great, do you think you could outline that in an issue? I fear I would not capture the need properly since I only understand the confusion and hopeful outcome but not the current reasons for why it is that way.

Reply:
https://github.com/elastic/kibana/issues/128364